AnnotatoBackend/Sources/App/DataAccess/DocumentSharesDataAccess.swift

```diff
@@ -0,0 +1,38 @@
+import FluentKit
+import AnnotatoSharedLibrary
+
+struct DocumentSharesDataAccess {
+    static func create(db: Database, documentShare: DocumentShare) async throws -> DocumentShare {
+        let existingDocumentShare = await findByDocumentIdAndRecipientId(db: db, documentShare: documentShare)
+
+        if let existingDocumentShare = existingDocumentShare {
+            throw AnnotatoError.modelAlreadyExists(
+                modelType: String(describing: DocumentShareEntity.self),
+                modelId: existingDocumentShare.documentId
+            )
+        }
```

Comment:
Why are we manually handling the error? Doesn't fluent throw an error because of the uniqueness constrait?

Reply:
If we manually handle it, we can customise the error message (it's like why we don't just allow DB errors to be passed all the way to the frontend)